### PR TITLE
Fix campaign advance crash

### LINKED ISSUE #3024: Crash on campaign advance
**Describe the bug**
Crash on campaign advance.

**To Reproduce**
1. Start an campaign
2. Use vcmiwin cheat
3. Continue campaign

**Expected behavior**
Next campaign scenario starts.

**Actual behavior**
Game crashes during sever shutdown because connection is empty. Same happens when you try to exit active game at any stage.

**Did it work earlier?**
Yes. 1.3.2 most likely it worked.

**Version**
 - OS: Windows, msvc
 - Version: latest develop, 1.4

### PR DESCRIPTION
**Pull request by Nordsoft91**
Fix #3024

### CODE REVIEW

File: client/CServerHandler.cpp [hunk 1]
```diff
@@ -924,7 +924,7 @@ void CServerHandler::threadHandleConnection()
 	try
 	{
 		sendClientConnecting();
-		while(c->connected)
+		while(c && c->connected)
```
Comment (nullkiller) [inline client/CServerHandler.cpp:927]: Does not it crash later?

Comment (nullkiller) [reply client/CServerHandler.cpp:927]: void CServerHandler::endGameplay calls connection a few times

Comment (Nordsoft91) [reply client/CServerHandler.cpp:927]: as I testes, was good. Will download build from PR later

Comment (Nordsoft91) [reply client/CServerHandler.cpp:927]: tested on build, everything is fine. Did you reproduce crash on PR build?

**Pull request merged**